src/auth/hasPermission.decorator.ts

```diff
@@ -0,0 +1,6 @@
+import { SetMetadata } from "@nestjs/common";
+import { permissionType } from "src/admin/schemas/admin.schema";
+
+export const PERMISSIONS_KEY = "permissions";
+export const HasPermissions = (...permissions: permissionType[]) =>
```

Comment:
what is this code mean? can you explain

Reply:
> what is this code mean? can you explain

This code defined a decorator, you can write '''@HasPermissions("super")''' above a router to let the reflector in role guard know, that this router needs to have a permission of role "super".